simpleParser.go

```diff
@@ -35,13 +42,24 @@ var ErrorNeedToStartNewCtx error = fmt.Errorf("Error: Need to spawn subcontext")
 var ErrorParenMismatch error = fmt.Errorf("Error: Parenthesis mismatch")
 var ErrorParenWSpace error = fmt.Errorf("Error: parenthesis must have white space before or after it")
 var NonErrorOneLayerDone error = fmt.Errorf("One layer has finished")
+var ErrorLeadingZeroes error = fmt.Errorf("Nested mode index must not have leading zeros")
+var ErrorAllInts error = fmt.Errorf("Nested mode index must be all integers")
+var ErrorEmptyNest error = fmt.Errorf("Nested mode bracket cannot be empty")
+var ErrorMissingBacktickBracket error = fmt.Errorf("Invalid field - could not find matching ending backtick or bracket")
+var ErrorEmptyLiteral error = fmt.Errorf("Literals cannot be empty")
 
 // Values by def should be enclosed within single quotes
 var valueRegex *regexp.Regexp = regexp.MustCompile(`^\".*\"$`)
 
 // Or Values can be int or floats by themselves (w/o alpha char)
 var valueNumRegex *regexp.Regexp = regexp.MustCompile(`^(-?)(0|([1-9][0-9]*))(\\.[0-9]+)?$`)
 
+// Field path can be integers
+var fieldTokenInt *regexp.Regexp = regexp.MustCompile(`[0-9]`)
+
+// But they cannot have leading zeros
+var fieldIndexNoLeadingZero *regexp.Regexp = regexp.MustCompile(`[0][0-9]+`)
```

Comment:
did you mean to exclude leading zero?  [^0][0-9]+ ?

Reply:
Yes fixed